sdk/azcore/progress.go

```diff
@@ -5,74 +5,61 @@
 
 package azcore
 
-import "io"
-
-// ********** The following is common between the request body AND the response body.
+import (
+	"io"
+)
 
 // ProgressReceiver defines the signature of a callback function invoked as progress is reported.
```

Comment:
Add a doc comment: Note that bytesTransferred resets to 0 if the stream is reset when retrying a network operation.